### PR TITLE
Add support for newly introduced CCM Rev3 on i.MX93

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
This series of changes will make sure that the CCM Rev3 driver from Zephyr is supported on i.MX93

### CODE REVIEW

**Comment by LaurentiuM1234** [comment]
@dleach02 @mmahadevan108 @danieldegrasse pls review.

File: mcux/mcux-sdk/devices/MIMX9352/MIMX9352_ca55_features.h [hunk 1]
```diff
@@ -550,5 +550,9 @@
 /* @brief WDOG_CNT can be 32-bit written. */
 #define FSL_FEATURE_WDOG_HAS_32BIT_ACCESS (1)
 
+#ifdef __ZEPHYR__
+#define FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL 1
+#endif /* __ZEPHYR__ */
+
```
Comment (mmahadevan108) [inline mcux/mcux-sdk/devices/MIMX9352/MIMX9352_ca55_features.h:556]: This should be passed in from the Zephyr side during compile.

File: mcux/mcux-sdk/devices/MIMX9352/drivers/fsl_clock.h [hunk 2]
```diff
@@ -1176,6 +1176,24 @@ static inline void CLOCK_SetRootClockMux(clock_root_t root, uint8_t src)
...
+{
+    CCM_Type *ccm_base = (CCM_Type *)regBase;
+
+    assert(src < 8U);
+    ccm_base->CLOCK_ROOT[root].CLOCK_ROOT_CONTROL.RW =
+        (ccm_base->CLOCK_ROOT[root].CLOCK_ROOT_CONTROL.RW & ~(CCM_CLOCK_ROOT_MUX_MASK)) | CCM_CLOCK_ROOT_MUX(src);
+    __DSB();
+    __ISB();
+}
+
```
Comment (mmahadevan108) [inline mcux/mcux-sdk/devices/MIMX9352/drivers/fsl_clock.h:1196]: This should be a PR to the SDK code. We do not want to maintain changes to SDK code in the Zephyr HAL repo.

**Comment by mmahadevan108** [comment]
@LaurentiuM1234, please work with the SDK team to update the SDK `fsl_clock` driver. We cannot maintain these changes inside the Zephyr HAL.

**Review by mmahadevan108 — CHANGES_REQUESTED** [review]
I feel most of the changes can be addressed by 
1. Working with the SDK team to update the clock driver
2. Adding changes to Zephyr to pass in the compile time defines. You can refer to the the MXRT code to see how this has been done.

**Comment by LaurentiuM1234** [comment]
> @LaurentiuM1234, please work with the SDK team to update the SDK `fsl_clock` driver. We cannot maintain these changes inside the Zephyr HAL.

These changes are based on the MIMX9352 support introduced by @JiafeiPan. This support was only merged in the NXP HAL and not the mcux_sdk repo. Without the support added to mcux_sdk repo I can't add the changes from this PR.

@JiafeiPan can you please add a PR to the mcux-sdk in which you add the support for MIMX9352?

**Comment by LaurentiuM1234** [comment]
@mmahadevan108 mcux-sdk PR https://github.com/nxp-mcuxpresso/mcux-sdk/pull/133 has been merged. Could we also merge this please?

**Comment by mmahadevan108** [comment]
The Update to SDK 2.14 should pull this in. PR to update to 2.14: https://github.com/zephyrproject-rtos/hal_nxp/pull/270

**Comment by mmahadevan108** [comment]
Please close this PR is it is addressed by the update to 2.14.

**Pull request closed without merging**